### PR TITLE
Feat: Pagination

### PR DESCRIPTION
**Pull request by kmaid**

### CODE REVIEW

File: apps/frontend/src/api/__tests__/util.test.tsx [hunk 1]
```diff
@@ -100,7 +100,9 @@ describe('useFetchOptions', () => {
 
 describe('createListApiUrl', () => {
   it('handles requests without parameters ', async () => {
-    expect(createListApiUrl('test').toString()).toMatch(/\/test$/);
+    expect(createListApiUrl('test').toString()).toMatch(
```
Comment (jeysal) [inline apps/frontend/src/api/__tests__/util.test.tsx:103]: come on, don't just go for what requires the least lines changed :see_no_evil: :D
it handles requests without parameters (checks the pathname ends with test as before)
it appends the default pagination params
it selects the given page

File: apps/frontend/src/hooks/__tests__/pagination.test.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    const { result } = renderHook(() => usePaginationParams(), {
+      wrapper: MemoryRouter,
+      initialProps: {
+        initialEntries: [{ search: '?currentPage=1' }],
+      },
+    });
+    expect(result.current.currentPage).toBe(1);
+  });
+
+  it('Returns provided provided page size', () => {
```
Comment (jeysal) [inline apps/frontend/src/hooks/__tests__/pagination.test.ts:25]: typo

File: apps/frontend/src/hooks/__tests__/pagination.test.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        current: { renderPageHref },
+      },
+    } = renderHook(() => usePagination(31, 10), {
+      wrapper: MemoryRouter,
+      initialProps: {
+        initialEntries: [{ search: '?currentPage=3' }],
+      },
+    });
+
+    expect(renderPageHref(0)).toMatchInlineSnapshot(`"?"`);
```
Comment (jeysal) [inline apps/frontend/src/hooks/__tests__/pagination.test.ts:73]: This is actually not so nice (browser keeps the `?`), I think `"."` may work in this case? (of course only if no other params are left, search params completely empty)

File: apps/frontend/src/hooks/__tests__/pagination.test.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+  it('Returns provided provided page size', () => {
+    const { result } = renderHook(() => usePaginationParams(12), {
+      wrapper: MemoryRouter,
+    });
+    expect(result.current.pageSize).toBe(12);
+  });
+});
+
+describe('usePagination', () => {
```
Comment (jeysal) [inline apps/frontend/src/hooks/__tests__/pagination.test.ts:33]: Only case coming to my mind that is not tested is the one for `numberOfItems === 0`, can we check that the calculation works for that?

**Review by jeysal — COMMENTED** [review]
Nice :+1:

**Review by fampinheiro — APPROVED** [review]

**Comment by jeysal** [comment]
RIP in conflicts. I'm sorry @kmaid, I saw this coming, but had no idea how much it'd be :see_no_evil: 
do you need help resolving? quick call or so

File: apps/frontend/src/api/__tests__/get-list.test.tsx [hunk 5]
```diff
@@ -16,17 +16,35 @@ jest.mock('../../config');
...
+    const url = createListApiUrl('test', { currentPage: 2, pageSize: 10 });
+    expect(url.searchParams.get('take')).toEqual('10');
+    expect(url.searchParams.get('skip')).toEqual('20');
+  });
+
   it('handles requests with a search query ', async () => {
     expect(
       createListApiUrl('test', { searchQuery: 'test123' }).toString(),
-    ).toContain('test?search=test123');
+    ).toContain('search=test123');
```
Comment (jeysal) [inline apps/frontend/src/api/__tests__/get-list.test.tsx:42]: I think one of the comments was not fully addressed here, can we check this properly using the URL's `searchParams`

**Review by jeysal — CHANGES_REQUESTED** [review]
This is blocked on the backend not working correctly yet as if I understand correctly it breaks even existing functionality (showing the first page).
When that is fixed and this is rebased, I'd like to see one tiny test change but looked over it again after the rebase and everything else looks fine :ok_hand:

**Comment by jeysal** [comment]
I'd approve, but still gives 400 Bad Request for me :/

**Comment by johnytiago** [comment]
BE to support this just landed 🎉

**Review by jeysal — APPROVED** [review]

**Pull request merged**